### PR TITLE
Enable and fixup findbugs

### PR DESCRIPTION
**Pull request by sul3n3t**
R: @alokmenghrajani 

- Enables findbugs in CI
- Changes LoginRequest to use AutoValue so char[] is not exposed
  - Subtle change makes the response type 400 when field missing, which
    is arguably more correct.
- Make some fields final
- Suppress TinyIntConverter warning where null should be returned
- Suppress checking jOOQ-generated files

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,4 +21,5 @@ script:
   - mvn test -P postgres
   - mvn test -P mysql
   - mvn test -P h2
+  - "mvn compile findbugs:check"
```
Comment (alokmenghrajani) [inline .travis.yml]: why are you using quotes

File: model/src/main/java/keywhiz/model/TinyIntConverter.java [hunk 2]
```diff
@@ -16,9 +16,11 @@
 
 package keywhiz.model;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import org.jooq.Converter;
 
 public class TinyIntConverter implements Converter<Byte, Boolean> {
+  @SuppressFBWarnings(value = "NP_BOOLEAN_RETURN_NULL", justification = "null is a valid DB value")
   @Override public Boolean from(Byte aByte) {
```
Comment (alokmenghrajani) [inline model/src/main/java/keywhiz/model/TinyIntConverter.java]: We could write this as `Converter<Byte, Optional<Boolean>>` and force callers to handle the `Optional.empty` case?

**Comment by alokmenghrajani** [comment]
:+1:

File: .travis.yml (hunk 1 above)
Comment (sul3n3t) [inline .travis.yml]: `:` looks like a yaml key/value

File: model/src/main/java/keywhiz/model/TinyIntConverter.java (hunk 2 above)
Comment (sul3n3t) [inline model/src/main/java/keywhiz/model/TinyIntConverter.java]: That didn't seem to work very well on CI.

**Pull request merged**